### PR TITLE
Reload inventory units in OrderShipping to make sure we have up-to-date states

### PR DESCRIPTION
**Pull request by philbirt**

### CODE REVIEW

**Comment by magnusvk** [comment]
Question: If we tackle the TODOs that are above and below this, might that obviate the need for this? I don't know enough to see why we need the reload, but that may be worth tackling if that's the case?

**Comment by jhawthorn** [comment]
I can see why this is necessary. We change the passed in set of `inventory_units`, which won't necessarily be the same ones as `shipment.inventory_units`: it could be `order.inventory_units` or `carton.inventory_units`. Basically, inventory_units are a mess.

However, I do think this needs a test.

File: core/app/models/spree/order_shipping.rb [hunk 1]
```diff
@@ -64,7 +64,7 @@ def ship(inventory_units:, stock_location:, address:, shipping_method:,
       # TODO: Remove tracking numbers from shipments.
       shipment.update_attributes!(tracking: tracking_number)
 
-      if shipment.inventory_units.all? {|iu| iu.shipped? || iu.canceled? }
+      if shipment.inventory_units(true).all? {|iu| iu.shipped? || iu.canceled? }
```
Comment (jordan-brough) [inline core/app/models/spree/order_shipping.rb]: Nitpick: `shipment.inventory_units.reload.all?` seems more readable to me.

**Comment by jordan-brough** [comment]
@magnusvk unfortunately I think we'd still need the `update_columns` line. It would just turn into a `shipment.ship!` line.

:+1: on the fix.

**Comment by magnusvk** [comment]
Cool, I'm convinced. :) :+1:

**Comment by adammathys** [comment]
This can be closed in favour of #170. Which is nearly identical. (It is rebased against master and has regression test.)

**Comment by jhawthorn** [comment]
closing in favour of #170

**Pull request closed without merging**